crest/Assets/Crest/Crest/Scripts/Underwater/UnderwaterRenderer.Mask.cs

```diff
@@ -162,61 +147,38 @@ internal void SetUpMaskTextures(RenderTextureDescriptor descriptor)
             descriptor.depthBufferBits = 0;
             descriptor.enableRandomWrite = true;
 
-            _maskRT = new RenderTexture(descriptor);
-            _maskRT.hideFlags = HideFlags.HideAndDontSave;
-            _maskRT.name = "_CrestOceanMaskTexture";
-            _maskTarget = new RenderTargetIdentifier
-            (
-                _maskRT,
-                mipLevel: 0,
-                CubemapFace.Unknown,
-                depthSlice: -1 // Bind all XR slices.
-            );
+            _maskRT.Release();
+            _maskRT.descriptor = descriptor;
 
             descriptor.colorFormat = RenderTextureFormat.Depth;
             descriptor.depthBufferBits = 24;
             descriptor.enableRandomWrite = false;
 
-            _depthRT = new RenderTexture(descriptor);
-            _depthRT.name = "_CrestOceanMaskDepthTexture";
-            _depthRT.hideFlags = HideFlags.HideAndDontSave;
-            _depthTarget = new RenderTargetIdentifier
-            (
-                _depthRT,
-                mipLevel: 0,
-                CubemapFace.Unknown,
-                depthSlice: -1 // Bind all XR slices.
-            );
+            _depthRT.Release();
+            _depthRT.descriptor = descriptor;
         }
 
         internal void SetUpVolumeTextures(RenderTextureDescriptor descriptor)
         {
+            if (!Helpers.RenderTargetTextureNeedsUpdating(_volumeFrontFaceRT, descriptor))
+            {
+                return;
+            }
+
             descriptor.msaaSamples = 1;
             descriptor.colorFormat = RenderTextureFormat.Depth;
             descriptor.depthBufferBits = 24;
 
-            Helpers.CreateRenderTargetTexture(ref _volumeFrontFaceRT, ref _volumeFrontFaceTarget, descriptor);
-            _volumeFrontFaceRT.name = "_CrestVolumeFrontFaceTexture";
+            _volumeFrontFaceRT.Release();
+            _volumeFrontFaceRT.descriptor = descriptor;
```

Comment:
Do you not have to call create, or does that happen elsewhere?

Reply:
We don't need to. It is done automatically when we use it with SetRenderTarget.